### DISCUSSION TITLE
No WebStorm IntelliSense for styled Ark components

### DISCUSSION THREAD

**Discussion by boar-is** [chakra-ui/panda#1787] [General]
### Description

Having the following code:
```tsx
import { styled } from '../styled-system/jsx';
import { ark } from '@ark-ui/react';

const Button1 = styled('button');

Button1({
  type:
});

const Button2 = ark.button;

Button2({
  type:
});

const Button3 = styled(ark.button);

Button3({
  type:
});
```

`Button3` has an intellisense ONLY for the panda properties (`fontWeight`, `css`, etc.), but not for the HTML properties (`type`, `aria-*`, etc.):
![image](https://github.com/chakra-ui/panda/assets/31354262/189e8ed1-5c05-4ee3-acf8-1f3c40ff4da7)

However, despite this, I still can't put arbitrary values to the HTML properties:
![image](https://github.com/chakra-ui/panda/assets/31354262/626b454f-f393-43d8-8be3-4f3acc2bf205)

`Button1` and `Button2` work as expected:
![image](https://github.com/chakra-ui/panda/assets/31354262/f36d3146-9e4c-463f-b48c-6ff63c836b05)
![image](https://github.com/chakra-ui/panda/assets/31354262/38e25e03-eac8-4578-95c0-465bf8ceac52)

The problem appears in WebStorm, when StackBlitz (Monaco/VSCode) looks fine.

### Link to Reproduction

https://stackblitz.com/edit/panda-intellisense?file=app%2Fbutton.tsx

### Steps to reproduce

The repro is a basic Next.js application with just Panda and Ark installed.

1. Clone the repro above
2. Open it via WebStorm
3. Go to `app/button.tsx`
4. Run intellisense dialog with `Ctrl+Space` for `Button3`
5. See the problem

### JS Framework

React (TS)

### Panda CSS Version

0.20.1

### Browser

N/A

### Operating System

- [ ] macOS
- [X] Windows
- [ ] Linux

### Additional Information

WebStorm 2023.2.5

Honestly, I don't know if I should address this issue here or to WebStorm (or maybe even to Ark). Let's start here.

**Comment by boar-is** [comment]
Also, Storybook render props are `any`.
```tsx
const meta = {
  component: Button,
} satisfies Meta<typeof Button>

export const RenderedButton = {
  render: (props) => <Button {...props} />
}
```
![image](https://github.com/chakra-ui/panda/assets/31354262/f4cf3a4d-fbf7-4cad-88ab-38c5062f6da9)
I also noticed that it takes some time for the intellisense to even infer `any`, like it consumes a lot of resources.

**Comment by astahmer** [comment]
this looks like a WebStorm specific issue, as Panda just use Typescript to get autocomplete suggestions

**Comment by cschroeter** [comment]
@astahmer 

Not sure, we also have a failing renovate PR for React types: https://github.com/chakra-ui/panda/pull/1719
Something is odd.

**Comment by segunadebayo** [comment]
I agree with @astahmer; this seems to be a Webstorm-specific issue. It works just fine in VSCode and StackBlitz

<img width="697" alt="CleanShot 2023-12-02 at 22 43 46@2x" src="https://github.com/chakra-ui/panda/assets/6916170/55e11b72-aa6c-4c3e-b63e-e0acd7be3e9c">

**Comment by segunadebayo** [comment]
@cschroeter

> Not sure, we also have a failing renovate PR for React types: #1719 Something is odd.

The failing PR has nothing to do with this issue. It's mostly related to the next.js link
<img width="1179" alt="CleanShot 2023-12-02 at 22 45 16@2x" src="https://github.com/chakra-ui/panda/assets/6916170/1401853d-4080-4ae3-be2b-3eb3105586cf">

**Comment by boar-is** [comment]
I reached out to JetBrains support with another but similar issue. They told me I should track [this issue](https://youtrack.jetbrains.com/issue/WEB-52943) and it's release is a "matter of weeks". I think it may fix this issue as well.
